### PR TITLE
feat: add PubSub provider removal doc

### PR DESCRIPTION
**Pull request by david-mcafee**
_Issue #, if available:_
https://github.com/aws-amplify/amplify-js/issues/7103

_Description of changes:_
Documenting [a recent update](https://github.com/aws-amplify/amplify-js/pull/7256) made by one of our open source contributors. Their PR addresses the following Amplify JS issue: https://github.com/aws-amplify/amplify-js/issues/7103

**NOTE: This PR should _NOT_ be merged until we have successfully shipped our next release (scheduled for 9/8/21)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/lib/pubsub/fragments/js/getting-started.md [hunk 1]
```diff
@@ -82,3 +82,30 @@ Amplify.addPluggable(new MqttOverWSProvider({
 ```
 
 You can integrate any MQTT Over WebSocket provider with your app. Click [here](https://docs.aws.amazon.com/iot/latest/developerguide/protocols.html#mqtt-ws) to learn more about MQTT Over WebSocket.
+
+## Removing a Provider
```
Comment (renebrandel) [inline docs/lib/pubsub/fragments/js/getting-started.md:86]:
```suggestion
## How to reconfigure PubSub provider during runtime
```

While this is technically accurate, the customer use case is likely as highlighted below in your text.

File: docs/lib/pubsub/fragments/js/getting-started.md [hunk 2]
```diff
@@ -82,3 +82,30 @@ Amplify.addPluggable(new MqttOverWSProvider({
...
+
+The `removePluggable(providerName: string)` function allows you to remove a PubSub provider by supplying the name of the provider.
+
+This is useful for: working with multiple concurrent PubSub providers, reconfiguring PubSub providers when changing authentication states, and changing the IoT connection region.
```
Comment (renebrandel) [inline docs/lib/pubsub/fragments/js/getting-started.md:90]:
```suggestion
Sometimes you need to reconfigure your PubSub provider when working with multiple concurrent PubSub providers, reconfiguring authentication states, or changing the IoT connection region. To reconfigure, the PubSub provider, remove the existing provider using `removePluggable` and add an updated PubSub provider using `addPluggable`.

The `removePluggable(providerName: string)` function allows you to remove a PubSub provider by supplying the name of the provider.
```

File: docs/lib/pubsub/fragments/js/getting-started.md [hunk 3]
```diff
@@ -82,3 +82,30 @@ Amplify.addPluggable(new MqttOverWSProvider({
...
+  aws_pubsub_region: '<YOUR-IOT-REGION>',
+  aws_pubsub_endpoint: 'wss://xxxxxxxxxxxxx.iot.<YOUR-IOT-REGION>.amazonaws.com/mqtt',
+}));
+
+// Apply second plugin with configuration
+pubsub.addPluggable(new MqttOverWSProvider({
+    aws_pubsub_endpoint: 'wss://iot.eclipse.org:443/mqtt',
+}));
+
+// Remove second plugin
```
Comment (renebrandel) [inline docs/lib/pubsub/fragments/js/getting-started.md:109]: This example only shows how to remove, which is nice but not the core use case highlighted in the section above. I'd recommend doing an example that shows something is being reconfigured.

File: docs/lib/pubsub/fragments/js/getting-started.md (hunk 1 above)
Comment (david-mcafee) [reply docs/lib/pubsub/fragments/js/getting-started.md]: Thanks, @renebrandel! Definitely agree with you on this.

File: docs/lib/pubsub/fragments/js/getting-started.md (hunk 3 above)
Comment (david-mcafee) [reply docs/lib/pubsub/fragments/js/getting-started.md:107]: Agree with you on this, I'll make that update now :)

File: docs/lib/pubsub/fragments/js/getting-started.md (hunk 2 above)
Comment (david-mcafee) [reply docs/lib/pubsub/fragments/js/getting-started.md]: I'll commit this and update - would like to remove the comma after "reconfigure".

File: docs/lib/pubsub/fragments/js/getting-started.md (hunk 1 above)
Comment (david-mcafee) [reply docs/lib/pubsub/fragments/js/getting-started.md]: nit: Perhaps we can update this to: `How to reconfigure PubSub providers during runtime`?

**Review by renebrandel — APPROVED** [review]

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Comment by tjleing** [comment]
Can you make these changes in `src/fragments/lib/pubsub/js/getting-started.mdx` instead/as well?  You will not see the update on the docs site if you only change files in the `/docs` folder, ever since the Next.js migration.  (We plan to delete the `/docs` folder in the next few days)

**Comment by tjleing** [comment]
If not that's okay too, we're going to do one final migration of MD->MDX content as well before deleting the old folders

**Comment by david-mcafee** [comment]
> Can you make these changes in `src/fragments/lib/pubsub/js/getting-started.mdx` instead/as well? You will not see the update on the docs site if you only change files in the `/docs` folder, ever since the Next.js migration. (We plan to delete the `/docs` folder in the next few days)

Sure thing! I'll push that change in a moment

**Review by tjleing — APPROVED** [review]
Looks great, thank you!

**Review by elorzafe — APPROVED** [review]
Thanks @david-mcafee 🌮 🥇

**Pull request merged**